### DISCUSSION TITLE
Welcome to SigLens Discussions!

### DISCUSSION THREAD

**Discussion by nkunal** [siglens/siglens#13] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by AndrewHess** [comment]
Hi I'm Andrew, I help develop and maintain the repository, mostly focusing on the backend

**Comment by amityahav** [comment]
Hey is there a blog regarding the architecture?

**Reply by nkunal** [reply]
We will add one soon

**Comment by timmyaditya** [comment]
Hi Community,

I have a folder that contains multiple log files how do I configure the paths of these log files/folder so that they can be live-tailed via the UI?

Also is it possible to index logs?

**Reply by nkunal** [reply]
yes you can index logs. we support various ingestion protocols: https://www.siglens.com/siglens-docs/category/log-ingestion/

One option for you to enable live tailing (aka ingesting these logs in real time) is to use agents like fluentd/fluennbit or SysLog remote-http-appender to get these logs ingested into SigLens (check above docs link). You can then view them in real time on our live-tail and/or search them in real time.

**Reply by timmyaditya** [reply]
Hi,

Thanks for the quick response!

I was trying to load mock data using the command given under the elastic search log ingestion option it gives an 400 error code stating
{"index":{"error":{"reason":"request body is required","type":"parse_exception"}},"status":400}

Is there anything I'm doing wrong?

**Reply by nkunal** [reply]
how did you send the mock data ? what cmd did you run ? It seems like your request did not contain a body hence it failed.
If you have ndjson formatted logfiles, then you can use our sigclient-tool to send data: , it takes a -x filePath option. https://github.com/sigscalr/sigscalr-client

**Reply by timmyaditya** [reply]
commands I tried.
curl -L https://github.com/sigscalr/ss-pub-data/releases/download/v1.0.0/prod-data-100mb.json.tar.gz -o prod-data-100mb.json.tar.gz 
tar -xjOf prod-data-100mb.json.tar.gz > prod-data-100mb.json  
curl http://localhost:8081/elastic/_bulk -i  --data-binary "@prod-data-100mb.json"

the client does work though.

**Reply by nkunal** [reply]
can you share a screenshot what cmds you tried and the failure you got ? (we tried the above cmds locally and they work)

**Reply by timmyaditya** [reply]
Here's the complete input and output:-

curl http://localhost:8081/elastic/_bulk -i --data-binary "@prod-data-100mb.json"
Warning: Couldn't read data from file "prod-data-100mb.json", this makes an 
Warning: empty POST.
HTTP/1.1 400 Bad Request
Server: SigLens-ingest
Date: Fri, 29 Dec 2023 19:10:20 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 95
Access-Control-Allow-Headers: Access-Control-Allow-Origin, Access-Control-Allow-Methods, Access-Control-Max-Age, Access-Control-Allow-Credentials, Content-Type, Authorization, Origin, X-Requested-With , Accept
Access-Control-Allow-Methods: HEAD, GET, POST, PUT, DELETE, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true

{"index":{"error":{"reason":"request body is required","type":"parse_exception"}},"status":400}

**Reply by nkunal** [reply]
ah do you have that `prod-data-100mb.json` file locally ? 
Did the previous tar cmd work and did it produce the above file ?

**Reply by timmyaditya** [reply]
yes!
